ansible/ansible-ai-connect-service
# allow custom models for all users after tech preview

Jira Issue: https://issues.redhat.com/browse/AAP-19483

## Description
For local/on-prem users, ability to specify a model id override should not be tied to a seat/commercial check. With this PR, if `ANSIBLE_AI_ENABLE_TECH_PREVIEW` is `True`, custom model overrides will remain seated user only. When `ANSIBLE_AI_ENABLE_TECH_PREVIEW` is `False`, any user can specify a custom model override. In a few weeks, after tech preview, we can remove the validation step entirely.

## Testing

### Steps to test
1. Pull down the PR
2. Run with ANSIBLE_AI_ENABLE_TECH_PREVIEW: True and confirm same behavior as today (model id overrides are seated user only)
3. Run with ANSIBLE_AI_ENABLE_TECH_PREVIEW: False and confirm any user can specify a model id.

### Scenarios tested
As described above, plus unit tests.

## Production deployment

- [x] This code change requires the following considerations before going to production: This PR renames the misleading "FREE" key and model env vars. Our secrets need to be updated to copy the values to ANSIBLE_WCA_HEALTHCHECK_API_KEY and ANSIBLE_WCA_HEALTHCHECK_MODEL_ID and later peel the old secrets out.
